### ISSUE TITLE
Don't throw an UndefinedClass error if a conditional established the class exists

### ISSUE THREAD

**Issue by dkarlovi** [vimeo/psalm#1801]
```php
<?php declare(strict_types = 1);

class C
{
	public function __construct()
	{
		if (class_exists(\Doesnt\Really::class)) {
			\Doesnt\Really::something(); // ERROR: UndefinedClass - 8:4 - Class or interface Doesnt\Really does not exist 
		}
	}
}
```

Executable: https://psalm.dev/r/08ac1bcb3e

**Comment by weirdan** [comment]
Works with literal string (as opposed to `::class` constant) though: https://psalm.dev/r/013d19a1f4
